# Теория вероятностей и математическая статистика
Задача 1:  Из колоды в 52 карты извлекаются случайным образом 4 карты. a)
Найти вероятность того, что все карты – крести. б) Найти вероятность, что
среди 4-х карт окажется хотя бы один туз.

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
  return int(factorial(n) / (factorial(k) * factorial(n-k)))

In [17]:
#a)
m = combinations(13, 4)
print(f'm= {m}')

m= 715


In [18]:
n = combinations(52, 4)
print(f'n = {n}')

n = 270725


In [19]:
P = m/n
print(f'P(clubs) = {round(P,4)}')

P(clubs) = 0.0026


In [13]:
#b)
n = combinations(52, 4)
print(f'n = {n}')

n = 270725


In [14]:
m = sum([combinations(4,1)*combinations(48,3),combinations(4,2)*combinations(48,2),combinations(4,3)*combinations(48,1),1])
print(f'm = {m}')

m = 76145


In [16]:
P = m/n
print(f'P(ace) = {round(P,4)}')

P(ace) = 0.2813


a) P = 0,0026 = 0,26%

b) P = 0,2813 = 28,13%

Задача 2: На входной двери подъезда установлен кодовый замок, содержащий
десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно
нажать одновременно. Какова вероятность того, что человек, не знающий код,
откроет дверь с первой попытки?

In [20]:
P = 1/combinations(10,3)
print(f'P = {round(P,4)}')

P = 0.0083


P = 0,0083 = 0,83%

Задача 3: В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий
случайным образом извлекает 3 детали. Какова вероятность того, что все
извлеченные детали окрашены?

In [21]:
P = combinations(9,3)/combinations(15,3)
print(f'P = {round(P,4)}')

P = 0.1846


P = 0.1846 = 18,46%

Задача 4: В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность
того, что 2 приобретенных билета окажутся выигрышными?

In [22]:
P = 1/combinations(100, 2)
print(f'P = {round(P, 4)}')

P = 0.0002


P = 0,0002 = 0,02%